src/components/TodoContext.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable */
+import React, {
+  Dispatch,
+  useEffect,
+  useReducer,
+} from "react";
+import { Todo } from "../types/Todo";
+import { getTodos } from '../api/todos';
+
+const USER_ID = 11384;
+
+export enum ACTIONS {
+  SORT,
+  SET_LENGTH,
+  SET_LIST,
+  SET_ERROR,
+}
+
+export enum FILTER {
+  ALL = 'All',
+  ACTIVE = 'Active',
+  COMPLETED = 'Completed',
+}
+
+type Action = { type: ACTIONS.SORT, payload: string }
+  | { type: ACTIONS.SET_LIST, payload: Todo[] }
+  | { type: ACTIONS.SET_LENGTH, payload: number }
+  | { type: ACTIONS.SET_ERROR, payload: string }
+
+interface Data {
+  list: Todo[],
+  sortBy: string,
+  totalLength: number,
+  error: string,
+};
+
+type State = [
+  state: Data,
+  dispatch: Dispatch<Action>,
+];
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case ACTIONS.SORT:
+      return [
+        {
+          ...state[0],
+          sortBy: action.payload,
+        },
+        () => { }
```

Comment:
you need to delete these empty callbacks `() => { }`

Reply:
have some troubles if deleting it )
<img width="1280" alt="Screenshot 2023-09-04 at 15 20 35" src="https://github.com/mate-academy/react_todo-app-loading-todos/assets/20277989/83d7d5cb-5224-4bb9-bfb5-cf6c18d4cb03">

Reply:
@haduigon then consider changing type of your State to be able to avoid passing empty callbacks all the time

Reply:
I' m trying to clarify how to do it properly )

Reply:
+